audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/request/AuditMessage.java

```diff
@@ -0,0 +1,230 @@
+        return numberOfRecordsReturned == that.numberOfRecordsReturned &&
+                numberOfRecordsProcessed == that.numberOfRecordsProcessed &&
+                timeStamp.equals(that.timeStamp) &&
+                serverIp.equals(that.serverIp) &&
+                serverHostname.equals(that.serverHostname) &&
+                context.equals(that.context) &&
+                Objects.equals(userId, that.userId) &&
+                Objects.equals(user, that.user) &&
+                Objects.equals(resourceId, that.resourceId) &&
+                Objects.equals(resource, that.resource) &&
+                Objects.equals(rules, that.rules) &&
+                Objects.equals(errorMessage, that.errorMessage);
+    }
+
+    @Override
+    @Generated
+    public int hashCode() {
+        return Objects.hash(timeStamp, serverIp, serverHostname, context, userId, user, resourceId, resource, rules, numberOfRecordsReturned, numberOfRecordsProcessed, errorMessage);
+    }
+
+    @Override
+    @Generated
+    public String toString() {
+        return new StringJoiner(", ", AuditMessage.class.getSimpleName() + "[", "]")
+                .add("timeStamp='" + timeStamp + "'")
+                .add("serverIp='" + serverIp + "'")
+                .add("serverHostname='" + serverHostname + "'")
+                .add("context=" + context)
+                .add("userId='" + userId + "'")
+                .add("user=" + user)
+                .add("resourceId='" + resourceId + "'")
+                .add("resource=" + resource)
+                .add("rules=" + rules)
+                .add("numberOfRecordsReturned=" + numberOfRecordsReturned)
+                .add("numberOfRecordsProcessed=" + numberOfRecordsProcessed)
+                .add("errorMessage='" + errorMessage + "'")
+                .add(super.toString())
+                .toString();
+    }
+
+    /**
+     * Builder class for the creation of AuditMessage instances.  This is a variant of the Fluent Builder
+     * which will use Java Objects or JsonNode equivalents for the components in the build.
+     */
+    public static class Builder {
+        private String timeStamp;
+        private String serverIp;
+        private String serverHostname;
+        private Context context;
+        private String userId;
+        private User user;
+        private String resourceId;
+        private LeafResource resource;
+        private Rules rulesApplied;
+        private long numberOfRecordsReturned;
+        private long numberOfRecordsProcessed;
+        private String errorMessage;
+
+
+        public static ITimeStamp create() {
+            return timeStamp -> serverIp -> serverHostname -> context -> userId -> user -> resourceId -> resource -> rules -> recordsReturned -> recordsApplied -> errorMessage ->
+                    new AuditMessage(timeStamp, serverIp, serverHostname, context, userId, user, resourceId, resource, rules, recordsReturned, recordsApplied, errorMessage);
+        }
```

Comment:
See the current AuditRequest message type for how you would have separate builders for each of the four classes

Reply:
Complete overhaul of AuditMessage. Have another look.